content/en/docs/tutorials/stateless-application/guestbook-logs-metrics-with-elk.md

```diff
@@ -0,0 +1,436 @@
+```
+### Install kube-state-metrics if needed
+
+```shell
+git clone https://github.com/kubernetes/kube-state-metrics.git kube-state-metrics
+kubectl create -f kube-state-metrics/kubernetes
+kubectl get pods --namespace=kube-system | grep kube-state
+```
+Verify that kube-state-metrics is running and ready
+```shell
+kubectl get pods -n kube-system -l k8s-app=kube-state-metrics
+```
+
+Output:
+```shell
+NAME                                 READY   STATUS    RESTARTS   AGE
+kube-state-metrics-89d656bf8-vdthm   2/2     Running     0          21s
+```
+## Clone the Elastic examples GitHub repo
+```shell
+git clone https://github.com/elastic/examples.git
+```
+
+The rest of the commands will reference files in the `examples/beats-k8s-send-anywhere` directory, so change dir there:
+```shell
+cd examples/beats-k8s-send-anywhere
+```
+
+## Create a Kubernetes Secret
+A Kubernetes {{< glossary_tooltip text="Secret" term_id="secret" >}} is an object that contains a small amount of sensitive data such as a password, a token, or a key. Such information might otherwise be put in a Pod specification or in an image; putting it in a Secret object allows for more control over how it is used, and reduces the risk of accidental exposure.
+
+{{< note >}}
+There are two sets of steps here, one for *self managed* Elasticsearch and Kibana (running on your servers or using the Elastic Helm Charts), and a second separate set for the *managed service* Elasticsearch Service in Elastic Cloud.  Only create the secret for the type of Elasticsearch and Kibana system that you will use for this tutorial.
+{{< /note >}}
+
+{{< tabs name="tab_with_md" >}}
+{{% tab name="Self Managed" %}}
+
+### Self managed
+Switch to the **Managed service** tab if you are connecting to Elasticsearch Service in Elastic Cloud.
+
+### Set the credentials
+There are four files to edit to create a k8s secret when you are connecting to self managed Elasticsearch and Kibana (self managed is effectively anything other than the managed Elasticsearch Service in Elastic Cloud).  The files are:
+
+1. ELASTICSEARCH_HOSTS
+1. ELASTICSEARCH_PASSWORD
+1. ELASTICSEARCH_USERNAME
+1. KIBANA_HOST
+
+Set these with the information for your Elasticsearch cluster and your Kibana host.  Here are some examples
+
+#### `ELASTICSEARCH_HOSTS`
+1. A nodeGroup from the Elastic Elasticseach Helm Chart:
+
+    ```shell
+    ["http://elasticsearch-master.default.svc.cluster.local:9200"]
+    ```
+1. A single Elasticsearch node running on a Mac where your Beats are running in Docker for Mac:
+
+    ```shell
+    ["http://host.docker.internal:9200"]
+    ```
+1. Two Elasticsearch nodes running in VMs or on physical hardware:
+
+    ```shell
+    ["http://host1.example.com:9200", "http://host2.example.com:9200"]
+    ```
+Edit `ELASTICSEARCH_HOSTS`
+```shell
+vi ELASTICSEARCH_HOSTS
+```
+
+#### `ELASTICSEARCH_PASSWORD`
+Just the password, no whitespace or quotes:
+
+    changeme
```

Comment:
The style guide for Kubernetes documentation recommends putting put sample text inside `<` and `>`.
We're keen (or should be keen) not to have our users sticking with default credentials.

I'd be tempted either to:
- explain in the text that the reader should put their own value in there
- have something like `<yoursecretpassword>` in place of `changeme`

Reply:
ok, where I have `Just the password, no whitespace or quotes` I will put `Just the password, no whitespace,  quotes, or <>` and make the example <yoursecretpassword>.  I suppose the same should be done for the username example.

Reply:
Hello @DanRoscigno. Couple more ideas:
nit: check spelling, s/Metricbet/Metricbeat
nit: Could you link up `Cleanup` emphasized at beginning to the Cleaning up header in the lower section, instead of emphasizing the text? Just put in a link.
nit: Yes, I think the title for this page could be different. The page is highlighting metric collection. Could the `What's Next` section refer to any other metrics-related info?
extra nit: The last output, showing everything deleted is a bit verbose. 
extra nit: I don't see the need to see the image of the Kibana dashboard, but others may agree that it is helpful.
extra nit: Why do you need to create a Secret?

Reply:
Thanks @kbhawkey 

* Corrected spelling, thanks!
* Removed cleanup from the objectives.
* I shortened the title and linked to pages related to metrics, troubleshooting, and logging.
* I removed the output from delete. 
* I will leave the image unless there is a problem with localization or some other issue.
* Secret: This is used for the authentication information to connect to the log and metric repository.  I generally try to avoid telling people to add this info to a configmap or environment variable.  If the recommended way to handle secrets is to use environment variables or put them in the manifest and then link to the Kubernetes Secrets docs then I could do that.

Thanks again!